components/brave_wallet_ui/components/desktop/wallet-page-wrapper/wallet-page-wrapper.tsx

```diff
@@ -65,27 +77,48 @@ export const WalletPageWrapper = (props: Props) => {
   const isWalletCreated = useSafeWalletSelector(WalletSelectors.isWalletCreated)
   const isWalletLocked = useSafeWalletSelector(WalletSelectors.isWalletLocked)
 
+  // State
+  const [headerShadowOpacity, setHeaderShadowOpacity]
+    = React.useState<number>(0)
+  const [headerDividerOpacity, setHeaderDividerOpacity]
+    = React.useState<number>(1)
+  const [headerHeight, setHeaderHeight] = React.useState<number>(0)
+
+  // Refs
+  let scrollRef = React.useRef<HTMLDivElement | null>(null)
+  const headerRef = React.createRef<HTMLDivElement>()
+
   // Computed
-  const showNavigationAndHeader =
-    isWalletCreated && !isWalletLocked &&
-    (
-      walletLocation.includes(WalletRoutes.PortfolioAssets) ||
-      walletLocation.includes(WalletRoutes.PortfolioNFTs) ||
-      walletLocation.includes(WalletRoutes.Accounts) ||
-      walletLocation.includes(WalletRoutes.Market) ||
-      walletLocation.includes(WalletRoutes.Activity) ||
-      walletLocation.includes(WalletRoutes.Send) ||
-      walletLocation.includes(WalletRoutes.Swap) ||
-      walletLocation.includes(WalletRoutes.FundWalletPageStart) ||
-      walletLocation.includes(WalletRoutes.DepositFundsPageStart)
-    )
 
   const headerTitle = AllNavOptions.find((option) =>
     walletLocation.includes(option.route))?.name ?? ''
 
-  const noCardPadding =
-    walletLocation === WalletRoutes.PortfolioAssets ||
-    walletLocation === WalletRoutes.PortfolioNFTs
+  React.useEffect(() => {
+    // Keeps track of the Header height to update
+    // the card top position and headers shadow.
+    setHeaderHeight(headerRef?.current?.clientHeight ?? 0)
+  }, [headerRef?.current?.clientHeight])
+
+  const onScroll = () => {
+    const scrollPosition = scrollRef.current
+    if (scrollPosition !== null) {
+      const { scrollTop } = scrollPosition
+      if (scrollTop === 0) {
+        setHeaderShadowOpacity(0)
+        setHeaderDividerOpacity(1)
+        return
+      }
+      if (scrollTop <= 64) {
+        setHeaderShadowOpacity((scrollTop / 8) * 0.01)
+        setHeaderDividerOpacity(
+          (100 - ((100 / 64) * scrollTop)) * 0.01
+        )
+        return
+      }
```

Comment:
Quite a few magics used here. Can you explain what's happening here as a comment? Is 8 related to 64, for example?

Reply:
Added comments to code!